File: src/main/java/com/codelog/lastlife/commands/CommandGiveLife.java
package com.codelog.lastlife.commands;

import com.codelog.lastlife.LifeContext;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class CommandGiveLife implements CommandExecutor {
    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (!(sender instanceof Player)) {
            sender.sendMessage("This is a player-only command!");
            return true;
        }

        if (args.length != 1) {
            sender.sendMessage(ChatColor.RED + "Incorrect number of arguments!");
            return false;
        }

        Player p = Bukkit.getServer().getPlayer(args[0]);
        if (p == null) {
            sender.sendMessage(ChatColor.RED + "No such player exists!");
            return true;
        }

        if (!LifeContext.get().hasPlayer(p.getUniqueId())) {
            sender.sendMessage(ChatColor.RED + "That player is no longer a part of the game!");
            return true;
        }

        Player senderPlayer = (Player)sender;
        if (LifeContext.get().getPlayerLives(senderPlayer.getUniqueId()) < 2) {
            sender.sendMessage(ChatColor.RED + "You do not have enough lives to give!");
            return true;
        }

        int receiverLives = LifeContext.get().getPlayerLives(p.getUniqueId());
        LifeContext.get().setPlayerLives(p.getUniqueId(), receiverLives + 1);
        int senderLives = LifeContext.get().getPlayerLives(senderPlayer.getUniqueId());
        LifeContext.get().setPlayerLives(senderPlayer.getUniqueId(), senderLives - 1);

        sender.sendMessage(ChatColor.BLUE + "You have successfully donated one life to " + ChatColor.YELLOW + p.getName());
        p.sendMessage(ChatColor.BLUE + "You have received a life from " + ChatColor.YELLOW + senderPlayer.getName());

        return true;
    }
}

File: src/main/kotlin/com/codelog/lastlife/commands/CommandGiveLife.kt
package com.codelog.lastlife.commands

import com.codelog.lastlife.Command
import com.codelog.lastlife.CommandDataPackage
import com.codelog.lastlife.LifeContext
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.TextColor
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.entity.Player

class CommandGiveLife(override val label: String = "givelife") : Command() {
    override fun execute(dataPackage: CommandDataPackage): Boolean {
        if (dataPackage.sender !is Player) {
            dataPackage.sender.sendMessage(Component.text("This is a player-only command!"))
            return true
        }

        val p: Player? = Bukkit.getServer().getPlayer(dataPackage.args[0])
        if (p == null) {
            dataPackage.sender.sendMessage(
                Component.text("No such player exists!", TextColor.color(0xff0000))
            )
            return true
        }

        if (dataPackage.args.size != 1) {
            dataPackage.sender.sendMessage(
                Component.text("Incorrect number of arguments!", TextColor.color(0xff0000))
            )
            return false
        }

        if (!LifeContext.hasPlayer(p.uniqueId)) {
            dataPackage.sender.sendMessage(
                Component.text("That player is no longer part of the game!", TextColor.color(0xff0000))
            )
            return true
        }

        val senderPlayer: Player = dataPackage.sender
        if (LifeContext.getPlayerLives(senderPlayer.uniqueId) < 2) {
            senderPlayer.sendMessage(
                Component.text("You don't have enough lives to give!", TextColor.color(0xff0000))
            )
            return true
        }

        val receiverLives = LifeContext.getPlayerLives(p.uniqueId)
        LifeContext.setPlayerLives(p.uniqueId, receiverLives + 1)
        val senderLives =  LifeContext.getPlayerLives(senderPlayer.uniqueId)
        LifeContext.setPlayerLives(senderPlayer.uniqueId, senderLives - 1)

        senderPlayer.sendMessage(ChatColor.BLUE.toString() + "You have successfully donated one life to " + ChatColor.YELLOW + p.name)
        p.sendMessage(ChatColor.BLUE.toString() + "You have received a life from " + ChatColor.YELLOW + senderPlayer.name)

        return true
    }
}